### PR TITLE
[feature/CORE-QA-1] Auth automation

### LINKED ISSUE #2: [CORE-QA-1] Auth automation

### PR DESCRIPTION
**Pull request by jonathan016**
Fixes #2

### CODE REVIEW

File: src/main/java/com/future/function/qa/api/auth/AuthAPI.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import io.restassured.specification.RequestSpecification;
+import net.thucydides.core.annotations.Step;
+
+public class AuthAPI extends BaseAPI {
+
+  @Step
+  public Response getLoginStatus(Cookie cookie) {
+
+    if (cookie == null) {
+      return getLoginStatusWithoutCookie();
```
Comment (oliverrsebastian) [inline src/main/java/com/future/function/qa/api/auth/AuthAPI.java:19]: can you remove the if statement? when the cookie is null, is it throw nullpointer?

File: src/main/java/com/future/function/qa/api/auth/AuthAPI.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    return base.cookie(cookie)
+        .delete();
+  }
+
+  @Step
+  @Override
+  public RequestSpecification prepare() {
+
+    base = super.prepare()
+        .basePath("/api/core/auth");
```
Comment (oliverrsebastian) [inline src/main/java/com/future/function/qa/api/auth/AuthAPI.java]: can this be moved as a constant?

**Review by oliverrsebastian — DISMISSED** [review]

File: src/main/java/com/future/function/qa/api/auth/AuthAPI.java (hunk 1 above)
Comment (jonathan016) [reply src/main/java/com/future/function/qa/api/auth/AuthAPI.java:19]: Unfortunately, it does (how annoying). Hence the if block

File: src/main/java/com/future/function/qa/api/auth/AuthAPI.java (hunk 2 above)
Comment (jonathan016) [reply src/main/java/com/future/function/qa/api/auth/AuthAPI.java]: Ok will move

**Review by oliverrsebastian — APPROVED** [review]

**Pull request merged**